python/tank/descriptor/descriptor_cached_config.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+from __future__ import with_statement
+
+import os
+
+
+from tank_vendor import yaml
+
+from . import constants
+from .errors import TankDescriptorError
+from .descriptor_config import ConfigDescriptor
+from .. import LogManager
+
+log = LogManager.get_logger(__name__)
+
+
+class CachedConfigDescriptor(ConfigDescriptor):
+    """
+    Descriptor that describes a cache Toolkit Configuration. Cached configurations are managed
+    by the ToolkitManager when bootstrapping and therefore are volatile.
+
+    These configurations live inside a bundle cache and have a core associated with them. They do
+    not share a core like an installed configuration would and therefore they have a set of
+    interpreter files that live alongside the configuration, even tough technically these files
+    are part of the core's configuration.
+    """
+    def _get_config_folder(self):
+        """
+        Returns the folder in which the configuration files are located.
+
+        :returns: Path to the configuration files folder.
+        """
+        self._io_descriptor.ensure_local()
+        return self._io_descriptor.get_path()
+
+    @property
+    def associated_core_descriptor(self):
```

Comment:
I think this should be renamed to `associated_core_descriptor_dict` and always return a dict, not potentially a uri string. (Just calling it `associated_core_descriptor` makes me think that it would return an object)